test/unit/org/apache/cassandra/io/filesystem/ForwardingFileSystem.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.filesystem;
+
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.FileSystem;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.WatchService;
+import java.nio.file.attribute.UserPrincipalLookupService;
+import java.nio.file.spi.FileSystemProvider;
+import java.util.Set;
+
+import com.google.common.collect.Iterables;
+
+public class ForwardingFileSystem extends FileSystem
+{
+    protected final FileSystem delegate;
+
+    public ForwardingFileSystem(FileSystem delegate)
+    {
+        this.delegate = delegate;
+    }
+
+    protected FileSystem delegate()
+    {
+        return delegate;
+    }
+
+    protected Path wrap(Path p)
+    {
+        return p;
+    }
+
+    protected Path unwrap(Path p)
+    {
+        return p;
+    }
+
+    @Override
+    public FileSystemProvider provider()
+    {
+        return delegate().provider();
```

Comment:
heh, IDEA's static analysis complains all over the place about these `delegate()` calls not being in try-with-resources blocks, but it isn't smart enough to realize it's equivalent to `delegate.provider()` and isn't actually creating anything :D

Could make the delegate private and use it directly internally w/ only the accessor available at protected, but whatever...

Reply:
I prefer this method as it allows subclasses to override... I actually used that during testing to make sure I knew what was touched and what wasn't

Reply:
k, I guess if you want to override the delegate on the fly, not just at construction time, makes sense